src/github_runner_manager/reactive/consumer.py

```diff
@@ -56,11 +96,75 @@ def consume(mongodb_uri: str, queue_name: str) -> None:
                     msg.reject(requeue=True)
                     raise JobError(f"Invalid job details: {msg.payload}") from exc
                 logger.info(
-                    "Received job with labels %s and run_url %s",
+                    "Received job with labels %s and job_url %s",
                     job_details.labels,
-                    job_details.run_url,
+                    job_details.job_url,
                 )
-                msg.ack()
+                _spawn_runner(
+                    runner_manager=runner_manager,
+                    job_url=job_details.job_url,
+                    msg=msg,
+                    github_client=github_client,
+                )
+
+
+def _spawn_runner(
+    runner_manager: RunnerManager, job_url: HttpUrl, msg: Message, github_client: GithubClient
+) -> None:
+    """Spawn a runner.
+
+    A runner is only spawned if the job has not yet been picked up by a runner.
+    After spawning a runner, it is checked if the job has been picked up.
+
+    If the job has been picked up, the message is acknowledged.
+    If the job has not been picked up after 5 minutes, the message is rejected and requeued.
```

Comment:
We should keep an eye out on whether this will cause a lot of re-queueing since if we use them for commonly used flavours, the chances that the exact job gets picked up is low. .g., if 10 jobs for `large` get submitted at the same time, the chances of a match between a job and runner is only 10% so we would have a 90% re-queueing rate which would mean that one reconciliation cycle is lost. Though that is likely mitigated by that some runner should be picking up those other jobs, unless we have a high queue length

Reply:
Perhaps another mitigation is if we see that a job is already picked up, we don't stop the script and instead go back to waiting for another job after acking the message. Both of these can be done in a follow up if we find this is a big issue

Reply:
I think this is something we need to monitor and discuss more internally. Thank you for your comments.